torch/csrc/jit/python/python_arg_flatten.cpp

```diff
@@ -65,6 +68,28 @@ void flatten_rec(PyObject* obj, ParsedArgs& args) {
     args.desc.structure.push_back(D::Variable);
   } else if (strcmp(THPUtils_typename(obj), "NoneType") == 0) {
     args.desc.structure.push_back(D::NoneType);
+  } else if (PyBool_Check(obj)) { // Wrap integers in bool tensors
+    at::Tensor tensor = scalar_to_tensor(at::Scalar(THPUtils_unpackBool(obj)));
+    PyObject* wappred_obj = THPVariable_Wrap(tensor);
+    auto& var = reinterpret_cast<THPVariable*>(wappred_obj)->cdata;
+    args.vars.push_back(var);
+    args.desc.metadata.emplace_back(var);
```

Comment:
what is metadata used for ?

Reply:
Haven't seen a use-case in unflatten or other places for these functions.